core/domain/exp_services_test.py

```diff
@@ -1305,6 +1305,49 @@ def test_unpublished_explorations_can_accept_commit_message(self):
             ), None)
 
 
+class UpdateExplorationContributionDatetimeTests(ExplorationServicesUnitTests):
```

Comment:
I would create a new test suite called FirstContributionDatetimeUpdateTests.

For setting up the exploration, look at core/test_utils for suitable methods in the superclass that you can make use of. There are other tests outside exp_services_test that set up explorations too...